### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by TheNilesh**
- Updated examples to demonstrate listing serverclasses and regions.  
- Added an example for automating kubeconfig token renewal with Terraform.  
- Removed the deprecated method for passing tokens to the provider.

### CODE REVIEW

File: docs/data-sources/kubeconfig.md
Comment (rajendragosavi) [inline docs/data-sources/kubeconfig.md:1]: Did you test this somewhere at least for syntax and parameter validation?

**Review by rajendragosavi — COMMENTED** [review]
Looks good to me. If this has been tested somewhere then it can go ahead.

**Review by ansalamdaniel — APPROVED** [review]
LGTM

File: docs/data-sources/kubeconfig.md
Comment (TheNilesh) [reply docs/data-sources/kubeconfig.md:1]: Thank you, Rajendra, for raising this. I’ve updated the lint target in the Makefile to ensure the syntax is valid. I found two indentation issues—though not blockers- those have been addressed.

The examples in the documentation have been manually verified by copying and using them without any modifications.

**Pull request merged**